### PR TITLE
TriggerPolicyUpdates after policy translation in k8s watcher

### PR DESCRIPTION
**Pull request by nebril**
Fixes: #1964

### CODE REVIEW

**Comment by ianvernon** [comment]
The build is stuck for some reason: https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1965/2/ . I've aborted it.

**Comment by nebril** [comment]
@ianvernon added some runtime tests, added you as a reviewer

**Review by ianvernon — APPROVED** [review]
Looks good. I think that we should have this merged as soon as possible given that this is a bug fix and adds test coverage. Please create another GitHub issue to create a Ginkgo version of this test. I will use this as the example for converting a test to Ginkgo in the upcoming demo I am giving to Cilium contributors on how to use Ginkgo.

**Comment by tgraf** [comment]
CI complains:
```
18:08:36 Error from server (BadRequest): error when creating "/home/vagrant/go/src/github.com/cilium/cilium/tests/k8s/tests/deployments/headless/policy.yaml": the object provided is unrecognized (must be of type ThirdPartyResourceData): version doesn't match, expecting: cilium.io/v1, got cilium.io/v2 ({"apiVersion":"cilium.io/v2"," ...)
```

**Comment by nebril** [comment]
@ianvernon #1985

File: tests/k8s/tests/04-toservices-test.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+log "creating policy, then endpoint, checking if ip is in toCIDR rules of endpoint"
+
+create_policy
+kubectl create -f "${headless_dir}/endpoint.yaml"
+wait_for_cilium_ep_gen k8s ${NAMESPACE} ${LOCAL_CILIUM_POD}
+
+epID=$(kubectl exec -n kube-system ${LOCAL_CILIUM_POD} cilium endpoint list | awk '$2=="Enabled" { print $1 }')
+log "endpoint ID: ${epID}"
+
+cidr=$(kubectl exec -n kube-system ${LOCAL_CILIUM_POD} cilium endpoint get ${epID} | jq '.[0]["policy"]["cidr-policy"]["egress"][0]' -M)
```
Comment (aanm) [inline tests/k8s/tests/04-toservices-test.sh]: The test failed because `jq` is not installed on the VM, can't you workaround it with grep?

**Review by aanm — APPROVED** [review]

File: tests/k8s/tests/04-toservices-test.sh (hunk 1 above)
Comment (tgraf) [reply tests/k8s/tests/04-toservices-test.sh]: I'm adding jq to the packer scripts

Comment (nebril) [reply tests/k8s/tests/04-toservices-test.sh]: Thanks!

Comment (nebril) [reply tests/k8s/tests/04-toservices-test.sh]: When will vms use new packer scripts?

Comment (tgraf) [reply tests/k8s/tests/04-toservices-test.sh]: I'm a bit confused now as `jq` was already part of the packer build scripts.

**Comment by tgraf** [comment]
```
10:46:56 Error from server (NotFound): error when deleting "/home/vagrant/go/src/github.com/cilium/cilium/tests/k8s/tests/deployments/headless/endpoint.yaml": endpoints "headless-service" not found
```

File: tests/k8s/tests/04-toservices-test.sh (hunk 1 above)
Comment (nebril) [reply tests/k8s/tests/04-toservices-test.sh]: @ianvernon can you shed some light on it?

**Comment by tgraf** [comment]
@nebril Needs a rebase

**Comment by aanm** [comment]
@nebril it seems the logs are big because we are printing this line https://github.com/cilium/cilium/blob/master/pkg/proxy/socket.go#L341 and this line 
https://github.com/cilium/cilium/blob/master/pkg/proxy/accesslog/log.go#L116
when doing the benchmarks

Also, the oxy/forward lines are being printed because we are setting up debug on the sirupsen package cc @raybejjani : https://github.com/cilium/cilium/blob/master/vendor/github.com/vulcand/oxy/forward/fwd.go#L164

**Comment by raybejjani** [comment]
I discussed with aanm the log being used by oxy and the best solution will be to not use the logrus package level logger, but to create our own instance. I've filed a ticket https://github.com/cilium/cilium/issues/2088 and assigned it to me but anyone can grab it if it's blocking you.

**Comment by ianvernon** [comment]
Something is borked with this PR for [this build](https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1965/30/), so I'm aborting it.

**Comment by ianvernon** [comment]
We cannot merge this in its current state. It is resulting in the creation of a huge number of logs (specifically from vulcand/oxy). When the tests fail, these logs are dumped to console (resulting in a console output of  almost 9 GB for a single run of this PR!). This is causing the Jenkins slaves to get stuck. This needs to be fixed and verified locally before trying to run another build of this PR.

**Comment by nebril** [comment]
@ianvernon sorry mate, I retriggered it before seeing your comment after seeing that CI is repaired. I will check it out locally tomorrow.

**Comment by nebril** [comment]
@ianvernon so can I actually run this on CI? I think the log issue was resolved by #2194?

**Pull request merged**